apps/molgenis-components/src/client/client.js

```diff
@@ -154,11 +155,58 @@ const request = async (url, graqphl) => {
   return result.data.data;
 };
 
-const cloneMetaData = (metaData) => {
-  // node js may not have structuredClone function, then fallback to deep clone via JSON
-  return typeof structuredClone === "function"
-    ? structuredClone(metaData)
-    : JSON.parse(JSON.stringify(metaData));
+const isFileValue = (value) => {
+  if (window && "File" in window) {
+    return value instanceof File;
+  } else {
+    throw "Files can only be uploaded via a browser client";
+  }
+};
+
+const toFormData = (rowData) => {
+  if (!FormData) {
+    throw "Files can only be uploaded via a browser client";
+  }
+  const formData = new FormData();
+  let nonFileValue = {};
+  let fileValues = {};
+
+  // split into file and non-file entries
+  for (const [key, value] of Object.entries(rowData)) {
+    isFileValue(value)
+      ? (fileValues[key] = value)
+      : (nonFileValue[key] = value);
+  }
+
+  // add the file objects to the formData and place a link to the object in the variables
+  for (const [key, value] of Object.entries(fileValues)) {
+    const id = Math.random().toString(36);
+    formData.append(id, value);
+    nonFileValue[key] = id;
+  }
+
+  formData.append("variables", JSON.stringify({ value: [nonFileValue] }));
+
+  return formData
+}
+
+const insertDataRow = (rowData, tableName, graphqlURL) => {
+  
+  const formData = toFormData(rowData);
+
+  const query = `mutation insert($value:[${tableName}Input]){insert(${tableName}:$value){message}}`
+  formData.append("query", query);
+
+  return axios.post(graphqlURL, formData);
+};
+
+const updateDataRow = (rowData, tableName, graphqlURL) => {
+  const formData = toFormData(rowData);
+
+  const query = `mutation update($value:[${tableName}Input]){update(${tableName}:$value){message}}`
+  formData.append("query", query);
+
+  return axios.post(graphqlURL, formData);
 };
 
 export { request };
```

Comment:
It would be nice to have the export at the top of the file so you can immediately see what the files does/exposes when you open it, and you can ctrl-click on the functions to see their implementations.

Reply:
done